connectors/sources/tests/test_confluence.py

```diff
@@ -5,16 +5,173 @@
+        {
+            "id": 4554779,
+            "name": "DEMO",
+            "_links": {
+                "webui": "/spaces/DM",
+            },
+        }
+    ],
+    "start": 0,
+    "limit": 1,
+    "size": 1,
+    "_links": {},
+}
+
+RESPONSE_PAGE = {
+    "results": [
+        {
+            "id": 4779,
+            "title": "ES-scrum",
+            "type": "page",
+            "history": {"lastUpdated": {"when": "2023-01-24T04:07:19.672Z"}},
+            "children": {"attachment": {"size": 2}},
+            "body": {"storage": {"value": "This is a test page"}},
+            "space": {"name": "DEMO"},
+            "_links": {
+                "webui": "/spaces/~1234abc/pages/4779/ES-scrum",
+            },
+        }
+    ],
+    "start": 0,
+    "limit": 1,
+    "size": 1,
+    "_links": {},
+}
+
+EXPECTED_PAGE = {
+    "_id": 4779,
+    "type": "page",
+    "_timestamp": "2023-01-24T04:07:19.672Z",
+    "title": "ES-scrum",
+    "body": "This is a test page",
+    "space": "DEMO",
+    "url": f"{HOST_URL}/spaces/~1234abc/pages/4779/ES-scrum",
+}
+
+EXPECTED_SPACE = {
+    "_id": 4554779,
+    "type": "Space",
+    "title": "DEMO",
+    "_timestamp": "2023-01-24T04:07:19+00:00",
+    "url": f"{HOST_URL}/spaces/DM",
+}
+
+RESPONSE_ATTACHMENT = {
+    "results": [
+        {
+            "id": "att3637249",
+            "title": "demo.py",
+            "type": "attachment",
+            "version": {"when": "2023-01-03T09:24:50.633Z"},
+            "extensions": {"fileSize": 230},
+            "_links": {
+                "download": "/download/attachments/1113/demo.py?version=1&modificationDate=1672737890633&cacheVersion=1&api=v2",
+                "webui": "/pages/viewpageattachments.action?pageId=1113&preview=demo.py",
+            },
+        }
+    ],
+    "start": 0,
+    "limit": 1,
+    "size": 1,
+    "_links": {},
+}
+
+EXPECTED_ATTACHMENT = {
+    "_id": "att3637249",
+    "type": "attachment",
+    "_timestamp": "2023-01-03T09:24:50.633Z",
+    "title": "demo.py",
+    "size": 230,
+    "space": "DEMO",
+    "page": "ES-scrum",
+    "url": f"{HOST_URL}/pages/viewpageattachments.action?pageId=1113&preview=demo.py",
+}
+
+RESPONSE_CONTENT = "# This is the dummy file"
+
+EXPECTED_CONTENT = {
+    "_id": "att3637249",
+    "_timestamp": "2023-01-03T09:24:50.633Z",
+    "_attachment": "IyBUaGlzIGlzIHRoZSBkdW1teSBmaWxl",
+}
+
+EXPECTED_BLOG = {
+    "_id": 4779,
+    "type": "blogpost",
+    "_timestamp": "2023-01-24T04:07:19.672Z",
+    "title": "demo-blog",
+    "body": "This is a test blog",
+    "space": "DEMO",
+    "url": f"{HOST_URL}/spaces/~1234abc/blogposts/4779/demo-blog",
+}
+
+EXPECTED_BLOG_ATTACHMENT = {
+    "_id": "att3637249",
+    "type": "attachment",
+    "_timestamp": "2023-01-03T09:24:50.633Z",
+    "title": "demo.py",
+    "size": 230,
+    "space": "DEMO",
+    "blog": "demo-blog",
+    "url": f"{HOST_URL}/pages/viewpageattachments.action?pageId=1113&preview=demo.py",
+}
+
+
+class MockJsonResponse:
```

Comment:
Could we use `AsyncMock` instead?

Reply:
We need json attribute and AsyncMock will not have a .json() attribute

Reply:
Updated, thanks [7880cf0](https://github.com/elastic/connectors-python/pull/604/commits/7880cf0cd4d5c1b14509a6b7659b7af65bc54f50)